[
  {
    "set": "cc",
    "id": 1,
    "messages": [
      {
        "msg": "usage",
        "id": 1,
        "description": "Usage message for the program. {0} is the program name.",
        "C": "Usage: {0} [file...]"
      },
      {
        "msg": "note",
        "id": 2,
        "description": "Label for a Note - extra information associated with a message",
        "C": "Note"
      },
      {
        "msg": "info",
        "id": 3,
        "description": "Label for an Info message - informational message which is not a warning/error",
        "C": "Information"
      },
      {
        "msg": "warning",
        "id": 4,
        "description": "Label for a Warning - code issue which isn't an error but can produce a 'code-smell'.",
        "C": "Warning"
      },
      {
        "msg": "error",
        "id": 5,
        "description": "Label for an Error - error in the code, but compilation can attempt to continue.",
        "C": "Error"
      },
      {
        "msg": "fatal_error",
        "id": 6,
        "description": "Label for a fatal error - an error in the code which stops compilation immediately.",
        "C": "Serious Error"
      },
      {
        "msg": "ice",
        "id": 7,
        "description": "Label for an internal compiler error - error in the compiler itself.",
        "C": "INTERNAL COMPILER ERROR"
      },
      {
        "msg": "command_line_file",
        "id": 8,
        "description": "Name to give the file that represents the command line.",
        "C": "(command line)"
      },
      {
        "msg": "standard_input_file",
        "id": 9,
        "description": "Name to give the file that represents the standard input",
        "C": "(standard input)"
      }
    ]
  },
  {
    "set": "errors",
    "id": 2,
    "messages": [
      {
        "msg": "too_many_errors",
        "id": 1,
        "description": "Too many errors have occured, stopping compilation. {0} is the number of errors.",
        "C": "Too many errors (have seen {0} error reports)."
      },
      {
        "msg": "file_error",
        "id": 2,
        "description": "Error whilst trying to read from a file, {0} is the file, {1} is the error message",
        "C": "File handling error in {0}: {1}"
      },
      {
        "msg": "missing_newline_at_end_of_file",
        "id": 3,
        "description": "Missing new line at end of non-empty file. (C2018-5.1.1.2-1-2)",
        "C": "Missing new line at end of non-empty file (C2018-5.1.1.2-1-2)."
      },
      {
        "msg": "splice_followed_by_end_of_file",
        "id": 4,
        "description": "We have a splice ('\\' followed by newline) immediately followed by end of file.  This is not permitted by C2018-5.1.1.2-1-2.",
        "C": "'\\' followed by newline, may not be the last characters in a file (C2018-5.1.1.2-1-2)."
      },
      {
        "msg": "malformed_character_80",
        "id": 5,
        "description": "We have a malformed UTF-8 character, shouldn't start with 0x80.",
        "C": "Malformed UTF-8 character.  Shouldn't start with top-two bits set to 0x80."
      },
      {
        "msg": "malformed_character_not_80",
        "id": 6,
        "description": "We have a malformed UTF-8 character, ran out of continuation characters too early.",
        "C": "Malformed UTF-8 character.  Not enough continuation characters before next character start."
      },
      {
        "msg": "malformed_character_early_line_end",
        "id": 7,
        "description": "We have a malformed UTF-8 character, ran out of continuation characters before the end of the line.",
        "C": "Malformed UTF-8 character.  Not enough continuation characters before the end of the line."
      }
    ]
  }
]
